cvat-core/src/math/index.js

```diff
@@ -105,10 +126,17 @@ function approximateAngleFactor(lines) {
   return { a, b };
 }
 
-function isFinite(point, infinityDistance) {
-  const { x, y } = point;
-  if (!infinityDistance) return true;
-  return Math.max(Math.abs(x), Math.abs(y)) < infinityDistance;
+function isFinite(vanishingPoint, points, thresholdAngle) {
+  if (!thresholdAngle) return true;
+  const angles = points.map(p => toPolarCoordinates(p, vanishingPoint).phi);
+  let maxAngle = 0;
+  for (let i = 0; i < angles.length - 1; i++) {
+    for (let j = i + 1; j < angles.length; j++) {
+      const a = getAngleBetween(angles[i], angles[j]);
```

Comment:
nit: you don't need to calculate max angle, can bail out when any angle is `> thresholdAngle`

Reply:
Agreed.